src/nix/substituters.rs

```diff
@@ -0,0 +1,90 @@
+//! This modules parses the get the substituters from the local nix install and the flake ref and combines them into a single list
+
+use crate::error::{Error, Result};
+
+pub type Substituters = Vec<String>;
+
+fn from_flake_ref(flake_ref: &str) -> Result<Substituters> {
+    let expr = format!(
+        "(import ((builtins.getFlake \"{}\").outPath + \"/flake.nix\")).nixConfig.extra-substituters or []",
+        flake_ref
+    );
```

Comment:
```suggestion
    let expr = "flakeRef: (import ((builtins.getFlake flakeRef).outPath + \"/flake.nix\")).nixConfig.extra-substituters or []";
```

and use `.args(["--argstr", flake_ref])` below

Reply:
Arrrgh yes, I thought about just that over the week-end and forgot to add that.

You can work around it with `nix eval --expr "{ flakeRef }: { result = blah; }.result" result` but that start s to be painfully involved :/

Reply:
I'll stick with the string interpolation right now, and try to switch to `nix-instantiate` after.